File: android/app/src/main/kotlin/com/thisissandipp/one/contacts/models/ContactEmail.kt
package com.thisissandipp.one.contacts.models

data class ContactEmail(
    val address: String? = "",
    val label: String? = "",
    val isPrimary: Boolean = false,
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "address" to address,
            "label" to label,
            "isPrimary" to isPrimary,
        )
    }
}

File: lib/contacts/domain/models/contact_email.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:one/utils/utils.dart';

part 'contact_email.freezed.dart';
part 'contact_email.g.dart';

@freezed
abstract class ContactEmail with _$ContactEmail {
  const factory ContactEmail({
    @Default('') String address,
    @Default('') String label,
    @Default(false) bool isPrimary,
  }) = _ContactEmail;

  factory ContactEmail.fromJson(JsonMap json) => _$ContactEmailFromJson(json);
}
